### PR TITLE
Add bulk interface

### PR DESCRIPTION
**Pull request by NickBehrends**

### CODE REVIEW

File: src/CSESoftware.Repository/IBulkRepository.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using CSESoftware.Core.Entity;
+
+namespace CSESoftware.Repository
+{
+    public interface IBulkRepository : IRepository
+    {
+        Task<IEnumerable<object>> CreateAsync<TEntity>(IEnumerable<TEntity> data) where TEntity : class, IBaseEntity;
```
Comment (TieDyeGeek) [inline src/CSESoftware.Repository/IBulkRepository.cs:9]: Should you make this return an IEnumerable<T> and specify that it is a TEntity<T>? Is that possible / does that make sense to do?

File: src/CSESoftware.Repository/Models/BatchQueries.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿using CSESoftware.Core.Entity;
```
Comment (TieDyeGeek) [inline src/CSESoftware.Repository/Models/BatchQueries.cs]: Reference to CSESoftware.Core.Entity not needed here

File: src/CSESoftware.Repository/IBulkRepository.cs (hunk 1 above)
Comment (NickBehrends) [reply src/CSESoftware.Repository/IBulkRepository.cs:9]: The intent here isn't so much to return the whole entities that were inserted, rather the id's so you can requery if you need or continue FK operations. if we do a select * on millions of records im not sure we even speed anything up with the overhead that would add.

File: src/CSESoftware.Repository/Models/BatchQueries.cs (hunk 2 above)
Comment (NickBehrends) [reply src/CSESoftware.Repository/Models/BatchQueries.cs]: Removed

**Review by jared-craig-cse — APPROVED** [review]

**Review by brickgw — APPROVED** [review]

**Pull request closed without merging**